woocommerce/woocommerce-android
# Login Enhancement: Update flow to allow site credentials login first before showing Jetpack missing screen

**Update**: This was originally put to Milestone 9.7, but is now moved to 9.6 in this PR. #6948

## Description

Fixes: #6879
Internal reference: pe5sF9-6C-p2

This modifies the existing login flow of:

1. "Enter your store address",
2. App tries to find Jetpack,
3.  Jetpack not found,
4. Jetpack required fragment is shown.

into:

1. "Enter your store address",
2. App tries to find Jetpack,
3.  Jetpack not found,
4. Login via site credentials fragment is shown,
5. Merchant logs in,
6. Jetpack required fragment is shown.

## Testing instructions

1. Start with a .org site with WooCommerce plugin but no Jetpack
2. Start app, choose "Enter your store address",
3. Enter site URL on the next screen,
4. The app will check and find that Jetpack is missing, and then show the site credentials login form,
5. Enter site username and password,
6. The app will show the Jetpack missing screen.

Flow should look like this:

| Enter site | Enter site credentials | Jetpack missing screen |
|-|-|-|
| <img src="https://user-images.githubusercontent.com/266376/178255350-ec34353d-4361-41de-8c4d-19fac575ba96.png" /> | <img src="https://user-images.githubusercontent.com/266376/178255335-636b0db5-40b9-4bc1-b339-bfe4e608d452.png" /> | <img src="https://user-images.githubusercontent.com/266376/178255356-032d938e-71ff-47e0-b919-c8e4efc63288.png" /> |
